Add possibility of having non-technical descriptive header in addition to technical header

kevinkhao opened the pull request.

kevinkhao committed:
[IMP] Add possibility of having non-technical descriptive header in addition to technical name header in imports and imports
Set default state of patterned import export as "pending" instead of "fail"

kevinkhao committed:
[IMP] Add functionality and tests for supporting m2o parent-child relationships

kevinkhao commented:
Also add support for adding parents and their children in one import

sebastienbeau committed:
[FIX] fix clear error col

sebastienbeau committed:
[REF] refactor code to avoid to inherit load method

sebastienbeau commented on pattern_import_export_xlsx/tests/test_pattern_import.py:
```diff
@@ -195,3 +207,11 @@ def disable_test_import_users_fail(self):
+
+    def test_import_partners_with_parents(self):
+        self._load_file("example.partners.parent.xlsx", self.ir_export_partner)
+        self.env.clear()
```
why to you use self.env.clear ?

kevinkhao replied:
Done. I presumed it was needed after seeing it once. Removed useless instances of it.

sebastienbeau commented on pattern_import_export/models/ir_exports.py:
```diff
@@ -227,6 +238,7 @@ def _generate_import_with_pattern_job(self, patterned_import):
         res = (
             self.with_context(load_format="flatty")
             .env[self.model_id.model]
+            .with_context(pattern_import_export_model=self.model_id.model)
```
There is already a context before, merge it

kevinkhao replied:
Done

kevinkhao committed:
[IMP] Remove useless self.env.clear() in tests

kevinkhao committed:
[IMP] Merge duplicate with_context()

kevinkhao committed:
[IMP] Separate patterned imports and exports into their own menu
Separate info field into info and info_separate for better tree view experience

sebastienbeau commented:
I have merged the PR without you last commit, please do a PR with this last commit (and the additionnal improvement)
